agent_app_sample_code/05_evaluate_poc_quality.py

```diff
@@ -0,0 +1,397 @@
+]
+
+overall_ratings = ["fail", "fail", "fail", "pass", "fail", "fail", "fail", "pass"]
+
+root_cause_rationales = [
+    "Retrieval quality is poor.",
+    "Retrieval quality is poor.",
+    "Response is grounded in retrieval, but retrieval is poor.",
+    "Relevant response grounded in the retrieved context and relevant, but retrieval is poor.",
+    "Hallucination",
+    "Hallucination",
+    "Good retrieval & grounded, but LLM does not provide a relevant response.",
+    "Good retrieval and relevant response. Collect ground-truth to know if the answer is correct.",
+]
+
+
+# Create new columns in the dataframe based on the conditions
+eval_results_df["overall/root_cause/rating"] = pd.Series(
+    pd.NA, index=eval_results_df.index
+)
+eval_results_df["overall/assessment"] = pd.Series(pd.NA, index=eval_results_df.index)
+eval_results_df["overall/root_cause/rationale"] = pd.Series(
+    pd.NA, index=eval_results_df.index
+)
+
+for condition, root_cause, overall_rating, root_cause_rationale in zip(
+    conditions, root_causes, overall_ratings, root_cause_rationales
+):
+    eval_results_df.loc[condition, "overall/root_cause/rating"] = root_cause
+    eval_results_df.loc[condition, "overall/assessment"] = overall_rating
+    eval_results_df.loc[
+        condition, "overall/root_cause/rationale"
+    ] = root_cause_rationale
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Determine the frequency of each root cause.
+
+# COMMAND ----------
+
+root_cause_frequencies = (
+    eval_results_df["overall/root_cause/rating"].value_counts(normalize=True) * 100
+)
+root_cause_rationale_frequencies = (
+    eval_results_df["overall/root_cause/rationale"].value_counts(normalize=True) * 100
+)
+
+print("Root cause frequencies: \n")
+print(root_cause_frequencies)
+print("\n-------\n")
+print("Root cause rationale frequencies: \n")
+print(root_cause_rationale_frequencies)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Review the traces to develop intuition about the issues
+
+# COMMAND ----------
+
+columns_to_display = [
+    "request_id",
+    "overall/root_cause/rating",
+    "overall/assessment",
+    "overall/root_cause/rationale",
+    "trace",
+    "expected_retrieved_context",
+    "expected_response",
+    "response/llm_judged/relevance_to_query/rating",
+    "response/llm_judged/relevance_to_query/rationale",
+    "response/llm_judged/groundedness/rating",
+    "response/llm_judged/groundedness/rationale",
+    # "response/llm_judged/correctness/rating",
+    # "response/llm_judged/correctness/rationale",
+    "retrieval/llm_judged/chunk_relevance/ratings",
+    "retrieval/llm_judged/chunk_relevance/rationales",
+    "retrieval/llm_judged/chunk_relevance/precision",
+]
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Results with a retrieval issue
+
+# COMMAND ----------
+
+filtered_df = eval_results_df[
+    eval_results_df["overall/root_cause/rating"] == "Improve Retrieval"
+][columns_to_display]
+if filtered_df.empty:
```

Comment:
Why not just call `display()` and let Databricks handle the UX here?

Reply:
`display()` breaks for this case 
![image](https://github.com/user-attachments/assets/31d1b402-da9e-484d-b916-d014eb223358)